services/storage/src/simcore_service_storage/db_file_meta_data.py

```diff
@@ -98,8 +98,8 @@ def _list_filter_with_partial_file_id_stmt(
         conditions.append(file_meta_data.c.file_id.startswith(file_id_prefix))
     if partial_file_id:
         conditions.append(file_meta_data.c.file_id.ilike(f"%{partial_file_id}%"))
-    if only_files:
-        conditions.append(file_meta_data.c.is_directory.is_(False))
+    if is_directory is not None:
+        conditions.append(file_meta_data.c.is_directory.is_(is_directory))
```

Comment:
the logic changes slightly here. I guess it's equivalent.

Reply:
It's more generic.

`None` == no-filter (both dirs and files)
`True` == only dirs
`False` == only files